reactor-core/src/test/java/reactor/core/publisher/TopicProcessorTest.java

```diff
@@ -834,4 +836,46 @@ private void assertProcessor(TopicProcessor<Integer> processor,
 		if (requestTaskExecutor != null)
 			assertEquals(requestTaskExecutor, processor.requestTaskExecutor);
 	}
+
+	@Test
+	public void serializedSinkSingleProducer() throws Exception {
+		TopicProcessor<Integer> processor = TopicProcessor.<Integer>builder()
+				.share(false)
+				.build();
+		FluxSink<Integer> sink = processor.sink();
+		assertTrue("Should not be serialized", sink instanceof SerializedSink);
```

Comment:
message should be "Should be serialized". Alternatively, use AssertJ: `Assertions.assertThat(sink).isInstanceOf(SerializedSink.class)`